README.md

```diff
@@ -44,11 +44,16 @@ This README documents the current strawman architecture.  Changes must be made v
 1. Components that do not fulfill the responsibilities of KPs and ARAs can still be stand-alone elements of the architecture to provide particular functionality; such tools will use the Translator ReasonerAPI whenever possible.
 1. Answer persistence will be the responsibility of the ARS.
 1. A system-wide UI will (eventually) exist, and will allow users to interpret answers, and reformulate questions.
-1. A Translator Registry will expose programmatically accessible metadata about KPs and ARAs, and will provide testing and reports as part of a continuous integration framework.
-    1. All KPs must be registered in the Translator Registry
-    1. KPs must expose machine-readable metadata describing the node and edge types that they provide, initially via a /predicates endpoint
-    1. KPs must expose machine-readable metadata describing the operations that they implement.  The method for exposing this metadata may be dependent upon the interface method (SmartAPI, ReasonerAPI, or KGX file).
-    1. Non-KP, Non-ARA components will also be collected in the registry, in a manner yet to be determined.
+1. The SmartAPI registry will serve as a Translator Registry, and will expose programmatically accessible metadata about KPs and ARAs.
+    1. All REST-Style SmartAPI KPs must be registered in the Translator Registry.
+    1. All Translator Reasoner API KPs must be registered in the Translator Registry. All metadata for Translator Reasoner APIs must be available via endpoints at the service, from which it will be extracted by the SmartAPI Registry.
+    1. All KGX files intended for graph transfer must be registered in the Translator Registry. All metadata for KGX files must be contained in associated metadata files, which will be consumed by the SmartAPI Registry.
```

Comment:
To register a KGX in SmartAPI registry, this is what we have discussed so far I believe:

* KGX should come with a companion API (Service Provider and SRI team are currently providing API-ification tools if needed)
* Register this API in SmartAPI registry in the same way as other KP APIs
* Provide a few extra metadata fields (to-be-finalized in [RegistryMetadata.md ](RegistryMetadata.md)) to indicate a KGX dump is available and the location of the KGX dump

We can discuss the above to see if we can reach the agreement with the architecture team.

Reply:
One can understand that a companion API for a KGX file of a knowledge graph makes the reuse of the Smart API registry a trivial exercise.  However, I wonder if the need to _host_ such an API (i.e. run a server exposing the web service) may otherwise be a unnecessary complication.  Many use cases for KGX file sharing involve ETL of the files, not their dynamic APi access (although that is fine as a option).

That said, one could simply envision a basic REST endpoint convention for retrieving the metadata file describing a KGX file-based knowledge graph, whose particulars can then be catalogued in the Registry in a simpler fashion, rather than a full SmartAPI spec.

Another component to facilitate KGX file access, at least, for graphs centrally hosted by Translator (if and when that happens) could be an active API (i.e. regular web service centrally online somewhere) to the catalog of knowledge graphs published as static KGX files which are archived in the designated Translator repository (wherever and whatever it may be).

Reply:
I want to clarify this point I mentioned above in my reply:

> KGX should come with a companion API (Service Provider and SRI team are currently providing API-ification tools if needed)

should be revised as this:

"A companion API for a KGX dump will be created with the API-fication tools provided from Service Provider and SRI."

This clarifies the overhead of the API-ification process will not be the responsibility of the KP teams producing KGX dumps.

Reply:
Tentatively, the survey of teams about their interest in sharing KGX formatted files of knowledge graphs indicated a common willingness to use a shared NCATS service to host such graphs online. In that case, NCATS could have an EC2 web server front end implementing a single common live API for accessing  those KGX files (as they are indexed) including serving any associated metadata in a standard fashion plus facilitating online streaming of such files (i.e. for ETL in various services) and perhaps(?!) providing some basic selective (filtered)  content access of such files(?)

Reply:
@RichardBruskiewich no objection to this plan, which adds a third-option to API-ify KGX. Exactly how to API-ifying KGX is something can evolve with the group over the time. I believe we have the consensus on these now:

* KGX should be also accessible via an API and registered at the SmartAPI registry. 
   option 1) one single KGX API instance, includes all KGX sources
   option 2) one KGX API instance per KGX source
   
* Arch group should come up one or multiple tools to API-ify KGX dumps, so this is not a responsibility/burden on a single KP team.

Reply:
Actually, both option 1 and 2 can be implemented concurrently, option 2 being for teams who are willing and earnest to manage their own KGX file archives (they can even choose to deploy the KGX API software we specify for the common NCATS KGX server, on their own server, for that purpose).